src/nm.c

```diff
@@ -327,6 +333,63 @@ write_bridge_params(const NetplanNetDefinition* def, GString *s)
     }
 }
 
+static void
+write_wireguard_params(const NetplanNetDefinition* def, GString *s)
+{
+    g_assert(def->tunnel.input_key);
+    g_string_append(s, "\n[wireguard]\n");
+
+    /* The key was already validated via validate_tunnel_grammar(), but we need
+     * to differentiate between base64 key VS absolute path key-file. And a base64
+     * string could (theoretically) start with '/', so we use is_wireguard_key()
+     * as well to check for more specific characteristics (if needed). */
+    if (def->tunnel.input_key[0] == '/' && !is_wireguard_key(def->tunnel.input_key)) {
+        g_fprintf(stderr, "%s: private key needs to be base64 encoded when using the NM backend\n", def->id);
+        exit(1);
+    } else
+        g_string_append_printf(s, "private-key=%s\n", def->tunnel.input_key);
+
+    if (def->tunnel.port)
+        g_string_append_printf(s, "listen-port=%u\n", def->tunnel.port);
+    if (def->tunnel.fwmark)
+        g_string_append_printf(s, "fwmark=%u\n", def->tunnel.fwmark);
+
+    for (guint i = 0; i < def->routes->len; i++) {
+        NetplanIPRoute *peer = g_array_index (def->routes, NetplanIPRoute*, i);
+        /* Routes which are not of "type: wireguard" are not for us */
+        if (!!g_ascii_strcasecmp(peer->type, "wireguard")) continue;
+
+        g_assert(peer->public_key);
+        g_string_append_printf(s, "\n[wireguard-peer.%s]\n", peer->public_key);
+
+        if (peer->keepalive)
+            g_string_append_printf(s, "persistent-keepalive=%d\n", peer->keepalive);
+        if (peer->via)
+            g_string_append_printf(s, "endpoint=%s\n", peer->via);
+        /* The key was already validated via validate_tunnel_grammar(), but we need
+         * to differentiate between base64 key VS absolute path key-file. And a base64
+         * string could (theoretically) start with '/', so we use is_wireguard_key()
+         * as well to check for more specific characteristics (if needed). */
+        if (peer->preshared_key) {
+            if (peer->preshared_key[0] == '/' && !is_wireguard_key(peer->preshared_key)) {
+                g_fprintf(stderr, "%s: shared key needs to be base64 encoded when using the NM backend\n", def->id);
```

Comment:
Similarly here. Maybe we should include information in the docs that using a keyfile path is only supported for the networkd backend?

Reply:
+1 I updated the docs.